demisto_sdk/commands/common/docker_helper.py

```diff
@@ -467,22 +483,56 @@ def _get_python_version_from_env(env: List[str]) -> Version:
 
 
 @functools.lru_cache
-def get_python_version(image: Optional[str]) -> Version:
-    log_prompt = f"Get python version from image {image}"
-    logger.debug(f"{log_prompt} - Start")
-    if not image or "pwsh" in image or "powershell" in image:
+def get_python_version(
+    image: Optional[str], should_pull_image: bool = True
+) -> Optional[Version]:
+    """
+    Get the python version of a docker image if exist.
+
+    Args:
+        image (str): the docker image
+        should_pull_image (bool): whether we should query the python version by pulling the image, True if yes, False if not.
+
+    Returns:
+        Version: Python version X.Y (3.7, 3.6, ..)
+    """
+    logger.debug(f"Get python version from image {image=}")
+
+    if not image:
         # When no docker_image is specified, we use the default python version which is Python 2.7.18
         logger.debug(
             f"No docker image specified or a powershell image, using default python version: {DEFAULT_PYTHON2_VERSION}"
         )
         return Version(DEFAULT_PYTHON2_VERSION)
-    if match := PYTHON_IMAGE_REGEX.match(image):
-        return Version(match.group("python_version"))
+
+    if "pwsh" in image or "powershell" in image:
+        logger.debug(
+            f"The {image=} is a powershell image, does not have python version"
+        )
+        return None
+
+    if python_version := DockerImagesMetadata.get_instance().python_version(image):
+        return python_version
+    logger.debug(
+        f"Could not get python version for {image=} from {DOCKERFILES_INFO_REPO} repo"
+    )
+
+    if python_version := _get_python_version_by_regex(image):
+        return python_version
+    logger.debug(f"Could not get python version for {image=} from regex")
+
+    if not should_pull_image:
+        logger.debug(f"get python version for {image=} from dockerhub api")
+        return _get_python_version_from_dockerhub_api(image)
+
     try:
+        logger.debug(
+            f"Getting python version from {image=} by pulling its image and query its env"
+        )
         return _get_python_version_from_image_client(image)
     except Exception:
         logger.debug(
-            "Could not get the python version from client. Trying with API",
+            f"Couldn't get the python version for image {image=} by pulling its image and query its env. Trying with API",
             exc_info=True,
         )
         return _get_python_version_from_dockerhub_api(image)
```

Comment:
if the DOCKERHUB_USER and DOCKERHUB_PASSWORD env variables are not set we could get rate limits.
However, if you login to the Docker with the PANW user, you will never get rate limited.

Reply:
@ilaner are you sure we won't get rate-limits even with `docker pull`? as far as i know you are limited in the number of images you can pull.

Reply:
btw it means that each time a person creates a content graph in local dev environment, one doesn't have  `DOCKERHUB_USER` and `DOCKERHUB_PASSWORD` configured, so in any case of misses it would just automatically pull the image.

Reply:
i think we should try catch for rate limit but still always try it first. Is there an issue with that @ilaner ? The potential loss is small (an api call) and the gain is saving a pull we dont need